Require reporting periods and form information; add them for San Diego.

bcipolli opened the pull request:
In order to parse more forms, we need to know what data are permanent (e.g. 460A) and what data are transient (e.g. Form 497). This also requires knowing the reporting periods, as transient data should be kept during the reporting period, and purged after it.

In this PR, I:
- Add methods for adding and using all that data.
- Add that data for San Diego (through May 22)
- Give a meaningful error message to indicate data downloads that fail to find the required reporting period / forms data: e.g. `Exception: You must add a reporting period for form 'Form 460 Schedule A', locality 'San Diego, CA', containing date 2016-04-22 00:00:00-07:00`
- Add documentation about how to give, and save, that data for other localities.

Commit:
ENH: expose Form and ReportingPeriod in the Django admin.

Commit:
ENH: add filing_deadline, locality, and permanent into ReportingPeriod. Use them!

Commit:
ENH: add 'force' capability to xformnetfilerawdata

Commit:
ENH: 460D

Commit:
ENH: add fixture for San Diego reporting periods, 2015 & 2016.

Commit:
BF: allow null filer ID when one's already been saved (cross-form interactions)

Commit:
ENH: delete rows of irrelevant data.

Commit:
BF: benefactor/beneficiary data for expenditures is in reversed fields; hack for 460D

Commit:
add fixture for latest form data from San Diego.

tdooner commented on finance/management/commands/xformnetfilerawdata.py:
```diff
@@ -273,22 +289,30 @@ def parse_ballot_info(row, locality, verbosity=1):
         # By ID
-        benefactor, _ = models.CommitteeBenefactor.objects.get_or_create(
+        benefactors = models.CommitteeBenefactor.objects.filter(
             filer_id=filer_id)
```
TODO: add unique constraint on filer_id on this table (?)

tdooner replied:
Then maybe just using `.get` would work. Though this glosess over non-numeric/pending values.

tdooner commented:
I read through this and didn't see anything too weird, you have my :+1: for whatever it's worth.

The amount of complexity in the xformnetfilerawdata seems like a good target for refactor.

tdooner committed:
Pretty print the JSON fixtures

It helps a lot when modifying them (clean git diffs) as well as just
making it easier to interact with as a human.

tdooner committed:
Add Oakland to Locality fixtures

There were some foreign key references to this in the ballot fixtures.

tdooner commented:
I took a stab at fixing the tests by adding some more fixture data, to no avail. I'm going to nonetheless merge this so I can build off it on develop.
